### PR TITLE
utils: cplay: Add support for ID3v2 tag skip

### PR DESCRIPTION
**Pull request by LaurentiuM1234**
This patch will allow playing MP3 files containing ID3v2 tag at the beginning of the file.

This is done simply by parsing the ID3v2 header, extracting the header size and seeking to that position (+10 bytes from the header size which is not included in the 'header size' field of the ID3v2 header)

Signed-off-by: Laurentiu Mihalcea <laurentiu.mihalcea@nxp.com>

### CODE REVIEW

**Comment by LaurentiuM1234** [comment]
@dbaluta @perexg @vinodkoul pls review

File: src/utils/cplay.c [hunk 1]
```diff
@@ -492,12 +495,94 @@ void get_codec_aac(FILE *file, struct compr_config *config,
...
+	/* read first BUFSZ chunk */
+	bytes_read = fread(buffer, sizeof(char), BUFSZ, file);
+
+	/* ID3v2 header is 10 bytes long
+	 *
+	 * if we can't read at least 10 bytes then there's obviously
+	 * no ID3v2 tag to skip.
+	 */
+	if (bytes_read < 10)
+		return 0;
```
Comment (dbaluta) [inline src/utils/cplay.c:548]: Please add a macro here for `10`. `#define ID3V2_HDR_SIZE 10`.

**Comment by dbaluta** [comment]
@LaurentiuM1234 minor comment added. Other than that it looks good to me.

File: src/utils/cplay.c (hunk 1 above)
Comment (LaurentiuM1234) [reply src/utils/cplay.c:548]: Fixed, thanks!

File: src/utils/cplay.c [hunk 2]
```diff
@@ -492,12 +495,94 @@ void get_codec_aac(FILE *file, struct compr_config *config,
...
+	 * 	=> 4 bytes long
+	 * 	=> the MSB of each byte is 0 so it needs to be ignored
+	 * 	when trying to parse the size.
+	 * 	=> this field doesn't include the 10 bytes which make up
+	 * 	the ID3v2 header so a +10 needs to be added to this
+	 * 	field in order to get the correct position of the end
+	 * 	of the ID3v2 tag.
+	 */
+
+	/* for now, we only support ID3v2 header found at beginning of
```
Comment (vinodkoul) [inline src/utils/cplay.c:530]: can we be consistent in styling please, we prefer the one you have used in previous comment block please

Comment (LaurentiuM1234) [reply src/utils/cplay.c:530]: Should be better now, thanks.

**Review by vinodkoul — CHANGES_REQUESTED** [review]
Just a nitpick otherwise lgtm

**Review by dbaluta — APPROVED** [review]

File: src/utils/cplay.c [hunk 3]
```diff
@@ -492,12 +495,97 @@ void get_codec_aac(FILE *file, struct compr_config *config,
 	codec->format = format;
 }
 
+
+static int skip_id3v2_header(FILE *file)
```
Comment (dbaluta) [inline src/utils/cplay.c:499]: Remove this new line. As we already have one above.

Comment (LaurentiuM1234) [reply src/utils/cplay.c:605]: Should be ok now, thanks

**Comment by LaurentiuM1234** [comment]
Update:

1) Solved merge conflict caused by https://github.com/alsa-project/tinycompress/pull/14 being merged. @vinodkoul @dbaluta I think this can be merged if there are no more comments.

**Review by dbaluta — APPROVED** [review]

**Comment by dbaluta** [comment]
@LaurentiuM1234 looks good to me.

@perexg @vinodkoul can you have a look?

File: src/utils/cplay.c [hunk 4]
```diff
@@ -599,12 +602,95 @@ void get_codec_aac(FILE *file, struct compr_config *config,
...
+	/* for now, we only support ID3v2 header found at the beginning
+	 * of the file so we need to move file cursor to the beginning
+	 * of the file
+	 */
+	ret = fseek(file, 0, SEEK_SET);
+	if (ret < 0)
+		return ret;
+
+	/* clean buffer for read operation */
+	memset(buffer, 0, BUFSZ);
```
Comment (perexg) [inline src/utils/cplay.c:645]: It's enough to have buffer size as ID3V2_HEADER_SIZE + 1 and set the trailing zero for strncmp() as `buffer[ID3V2_HEADER_SIZE] = '\0'` here.

Also, you can replace BUFSZ with `sizeof(buffer)`.

**Review by perexg — CHANGES_REQUESTED** [review]

File: src/utils/cplay.c (hunk 4 above)
Comment (LaurentiuM1234) [reply src/utils/cplay.c]: Thanks, should be better now. Also fixed magic number in the `strncmp().` Should be a bit clearer now.

**Pull request merged**